drivers/hmis/app/graphql/mutations/delete_client.rb

```diff
@@ -0,0 +1,74 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class DeleteClient < BaseMutation
+    argument :id, ID, required: true
+    argument :confirmed, Boolean, 'Whether warnings have been confirmed', required: false
+
+    field :client, Types::HmisSchema::Client, null: true
+
+    def resolve(id:, confirmed: false)
+      client = Hmis::Hud::Client.find_by(id: id)
+
+      warnings, resolvable_enrollments = check_enrollments(client, ignore_warnings: confirmed)
+
+      return { client: nil, errors: warnings } if warnings.present?
+
+      default_delete_record(
+        record: client,
+        field_name: :client,
+        permissions: :can_delete_clients,
+        after_delete: -> do
+          resolvable_enrollments.each do |enrollment|
+            enrollment.household_members.where.not(personal_id: client.personal_id).first&.update!(relationship_to_ho_h: 1)
```

Comment:
The db changes (deletion & relationship changes) should happen in a transaction. If any of them fail, the whole operation should fail (and none of the changes should be persisted).

We haven't been consistent about that in all the mutations so far, but I have been going back and adding transactions where needed.

Reply:
I just wrapped the whole thing in a block with `client.transaction do`, which I believe should accomplish this, but please tell me if I'm mistaken about that.